contrib/jmx-metrics/src/main/java/io/opentelemetry/contrib/jmxmetrics/GroovyRunner.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.contrib.jmxmetrics;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import groovy.lang.Binding;
+import groovy.lang.GroovyShell;
+import groovy.lang.Script;
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.codehaus.groovy.control.CompilationFailedException;
+
+public class GroovyRunner {
+  private static final Logger logger = Logger.getLogger(GroovyRunner.class.getName());
+
+  private final GroovyShell gshell = new GroovyShell();
+
+  private final Script script;
+  private final GroovyUtils gutil;
+
+  GroovyRunner(final String groovyScript, final JmxClient jmxClient, final GroovyUtils gutil) {
```

Comment:
Looks like we don't handle dynamic updates to the file - this is probably safer for groovy anyways. But it'd be nice in the future if we can handle a yaml file with dynamic updates.